bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/osgi/AbstractAsyncBundleProcessor.java

```diff
@@ -107,16 +107,16 @@ protected boolean isBundleRelevant(Bundle bundle) {
     }
 
     /**
-     * Checks for the existence of a given directors inside the bundle.
+     * Checks for the existence of a given directors inside the bundle and its attached fragments.
      *
      * Helper method which can be used in {@link #isBundleRelevant(Bundle)}.
      * 
      * @param bundle
      * @param path the directory name to look for
-     * @return <code>true</code> id the bundle contains the given directory
+     * @return <code>true</code> id the bundle or one of its attached fragments contain the given directory
      */
     protected final boolean isDirectoryPresent(Bundle bundle, String path) {
-        return bundle.getEntry(path) != null;
+        return bundle.findEntries(path, "*", true) != null;
```

Comment:
The XmlDocumentBundleTracker uses that call in the constructor <code>super(bundleContext, Bundle.ACTIVE, null);</code>.

```text
	 * @param stateMask The bit mask of the {@code OR}ing of the bundle states
	 *        to be tracked.
```

So, if the state mask is "Bundle.ACTIVE", I see no prob. But perhaps you should add it as an comment.

```text
If this bundle's state is INSTALLED, this method must attempt to resolve this bundle before attempting to get the specified resource.
```

Is the state is installed, the resolving would be done by the getResource method (or at least tried to be done). Is this correct?